File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
  unordered_map<char, int> mp;
  int i = 0, result = 0;

  for (int j = 0; j < s.length(); j++) {
    mp[s[j]]++;

    while (mp[s[j]] > 1) {
      mp[s[i]]--;
      i++;
    }

    result = max(result, j - i + 1);
  }

  return result;
}

};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int traverse(TreeNode root, int maxSoFar) {
        if (root == null) {
            return 0;
        }

        int count = 0;

        if (root.val >= maxSoFar) {
            count = 1;
        }

        maxSoFar = Math.max(maxSoFar, root.val);

        count += traverse(root.left, maxSoFar);
        count += traverse(root.right, maxSoFar);

        return count;
    }
    public int goodNodes(TreeNode root) {
        if (root == null) {
            return 0;
        }
        List<Integer> inorderList = new ArrayList<>();
       return traverse(root, root.val);
    }
}
